packages/core-ethereum/src/indexer/index.ts

```diff
@@ -86,9 +87,30 @@ class Indexer extends EventEmitter {
     this.chain.subscribeBlock((b) => {
       this.onNewBlock(b)
     })
+
     this.chain.subscribeError((error: any) => {
       log(chalk.red(`etherjs error: ${error}`))
-      this.restart()
+      // if provider connection issue
+      console.log('code', error.code, errors.SERVER_ERROR)
+      if (
+        [error?.code, String(error)].includes(errors.SERVER_ERROR) ||
```

Comment:
How does this cope with the unhandled promise rejection with the error `ECONNRESET`?

Reply:
Follow up question on `ECONNRESET`, should the `core-ethereum` try to connect to a new provider or is it enough to simply retry calling the function with backoff?

Reply:
We don't have a concept for multiple provider endpoints yet. Also the error isn't descriptive enough to identify whether this is a temporary or permanent issue. Thus, I would stick to the retry for now.